lib/sqlalchemy_ingres/base.py

```diff
@@ -376,8 +376,10 @@ def get_primary_keys(self, connection, table_name, schema=None, **kw):
         
         try:
             rs = connection.exec_driver_sql(sqltext, params)
-            
-            return [row[0].rstrip() for row in rs.fetchall()]
+
+            cols = [row[0].rstrip() for row in rs.fetchall()]
+            return {"constrained_columns": [] if cols is None else cols, "name": None}
```

Comment:
How does returning a dict (instead of a list) work for sqlalchemy 1.x?

Reply:
Regarding how SQLAlchemy 1.x works when returning a dictionary, I was able to successfully run the two Python test scripts in this PR after downgrading the `langchain` and `langchain-community` packages as per #32 due to the noted langchain bug.

Both test cases pass with these package versions (in additional to already passing with SQLAlchemy `2.0.27`).

    langchain           0.1.6
    langchain-community 0.0.19
    SQLAlchemy          1.4.51

Reply:
I ran the SQLAlchemy test suite with and without this PR fix, using SQLAlchemy 2.0.27 and 1.4.51. In each case, the test suite execution improved with the fix in place, with more tests that passed, fewer that failed, and fewer errors.

### SQLAlchemy 2.0.27 Test suite execution results

    langchain           0.1.6
    langchain-community 0.0.19
    langchain-core      0.1.23
    pyodbc              5.1.0
    pypyodbc            1.3.6
    SQLAlchemy          2.0.27
    sqlalchemy-ingres   0.0.5.dev1

Feb 16 | Passed | Failed | Errors
--|--|--|--
Without Fix | 12329 | 1951 | 2250
With Fix | 12385 | 1901 | 2239
Improvement | +56 | -50 | -11

Feb 19 | Passed | Failed | Errors
--|--|--|--
Without Fix | 12339 | 1949 | 2224
With Fix | 12401 | 1901 | 2219
Improvement | +62 | -48 | -5

***  

### SQLAlchemy 1.4.51 Test suite execution results

    langchain           0.1.6
    langchain-community 0.0.19
    langchain-core      0.1.23
    pyodbc              5.1.0
    pypyodbc            1.3.6
    SQLAlchemy          1.4.51
    sqlalchemy-ingres   0.0.5.dev1

Feb 19 | Passed | Failed | Errors
--|--|--|--
Without Fix | 12329 | 1952 | 2242
With Fix | 12393 | 1902 | 2238
Improvement | +64 | -50 | -4